PROD-464 Stack of daily decks

neel-ds opened the pull request:
- Description
It adds a stack of daily decks, enabling access to past daily decks on Stacks page.

- What are the steps to test that this code is working?
You can find Daily Decks Stack on top of list.

- Screen shots or recordings for UI changes
<img width="576" alt="Screenshot 2024-11-14 at 6 40 59 PM" src="https://github.com/user-attachments/assets/37a8fd76-af32-41a2-b5ba-cc4c48273d70">

neel-ds commented on app/stacks/daily-deck/page.tsx:
```diff
@@ -0,0 +1,92 @@
+          />
+        </div>
+        <div className="flex flex-col">
+          <h1 className="text-base mb-3">Daily Deck</h1>
```
```suggestion
          <h1 className="text-base mb-3">Daily Decks</h1>
```

samott commented:
Looks good.

Not a huge fan of that `Number()` conversion map (I think the consumer could do it) but it's an issue with the existing code - maybe to revisit another time.

neel-ds commented:
> Looks good.
> 
> Not a huge fan of that `Number()` conversion map (I think the consumer could do it) but it's an issue with the existing code - maybe to revisit another time.

Could you please confirm if my latest commit addresses the feedback?

samott commented on app/stacks/daily-deck/page.tsx:
```diff
@@ -0,0 +1,92 @@
+                .map((cr) => ({
+                  ...cr,
+                  rewardTokenAmount: Number(cr.rewardTokenAmount),
+                })) as (Omit<ChompResult, "rewardTokenAmount"> & {
```
@neel-ds I was thinking instead of doing this `map()` call, The `Number()` cast could happen inside the `reduce()` call in StackDeckCard.tsx

neel-ds replied:
Made the changes in the recent commit. Can you please review and let me know if I am missing something?

samott replied:
Looks good to me!

samott commented:
> > Looks good.
> > Not a huge fan of that `Number()` conversion map (I think the consumer could do it) but it's an issue with the existing code - maybe to revisit another time.
> 
> Could you please confirm if my latest commit addresses the feedback?

Your other change makes sense, but see my other comment for clarification.

marvinmarnold reviewed (changes requested):
Stack page link not working

![Screenshot 2024-11-18 at 12 25 37 PM](https://github.com/user-attachments/assets/16133d8b-19f7-4d9a-81db-65414b0eec17)

marvinmarnold commented:
> Stack page link not working

I think this is the corresponding error: https://gator-labs.sentry.io/issues/6073321589/?alert_rule_id=15396170&alert_type=issue&notification_uuid=4eaee4f7-34e4-4838-bc2a-dc285c56e6b8&project=4507686429130752&referrer=slack

neel-ds commented:
> https://gator-labs.sentry.io/issues/6073321589/?alert_rule_id=15396170&alert_type=issue&notification_uuid=4eaee4f7-34e4-4838-bc2a-dc285c56e6b8&project=4507686429130752&referrer=slack

This made it easy. Thank you! I fixed the issue. If there are no daily decks then it won't throw above error anymore.
